stan/math/prim/mat/prob/multi_normal_cholesky_lpdf.hpp

```diff
@@ -101,37 +102,63 @@ typename return_type<T_y, T_loc, T_covar>::type multi_normal_cholesky_lpdf(
     check_not_nan(function, "Random variable", y_vec[i]);
   }
 
+  operands_and_partials<T_y, T_loc, T_covar> ops_partials(y, mu, L);
+
   if (size_y == 0)
-    return lp;
+    return ops_partials.build(0.0);
 
   if (include_summand<propto>::value)
-    lp += NEG_LOG_SQRT_TWO_PI * size_y * size_vec;
+    logp += NEG_LOG_SQRT_TWO_PI * size_y * size_vec;
+
+  matrix_partials_t L_dbl = value_of(L);
+  // do a single inversion
+  matrix_partials_t inv_trans_L_dbl
+      = mdivide_left_tri<Eigen::Upper>(transpose(L_dbl));
 
-  if (include_summand<propto, T_covar_elem>::value)
-    lp -= L.diagonal().array().log().sum() * size_vec;
+  // analytic expressions taken from
```

Comment:
These should go in top-level doc.

Reply:
Are u fine with just putting the link to the top-level or are u asking me to include the math expressions into the file?

Reply:
A link's fine.